### DISCUSSION TITLE
Is Katex HTML not backwards compatible?

### DISCUSSION THREAD

**Discussion by AhmedFarhan252** [KaTeX/KaTeX#3577] [Q&A] [answered]
It seems that re-using Katex HTML (from `renderToString`) that was generated in an old version of Katex causes some rendering issues when used with a newer version of Katex. This issue occurred after updating from v0.6 to v0.15.2 specifically.

**Steps to Reproduce:**
1. Get Katex HTML from `renderToString` on the old version.
2. Store this HTML somewhere e.g. backend / local storage
3. Update Katex to newer version.
4. Now, fetch and render the stored Katex HTML. 
5. Katex HTML seems to be messed up.

For reference, here is the rendered Katex on the old version: 
![image](https://user-images.githubusercontent.com/47603421/157504942-950fe4d4-7c07-4834-adc3-cbcaccc2c354.png)

and the same Katex rendered after the version update:
![image](https://user-images.githubusercontent.com/47603421/157505612-1c634c4b-f95f-4114-9e05-3877b150a8ff.png)

**Comment by edemaine** [comment, marked as answer]
This is correct. The CSS classes have changed occasionally, and you need to use the KaTeX CSS version that matches the KaTeX JavaScript version (from when KaTeX was run).

**Reply by AhmedFarhan252** [reply]
Thank you for the prompt response. 

So essentially, one would need to regenerate the Katex HTML on the newer version for it to be properly rendered.

**Reply by edemaine** [reply]
Yes, or use the old CSS (and only use the old KaTeX compiler).
